### ISSUE TITLE
Double quotes escaping not recognized?

### ISSUE THREAD

**Issue by matthewpersico** [koalaman/shellcheck#1227]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1078, SC1079
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
holdstdout=dummy
holdstderr=dummy
profilelog=dummy
alllog=dummy
trap "eval ""exec 1>&$holdstdout $holdstdout>&-""      # Restore stdout and close file descriptor.
      eval ""exec 2>&$holdstderr $holdstderr>&-""      # Restore stderr and close file descriptor.
      [ -f $profilelog ] && cat $profilelog
      [ -f $alllog ]     && cat $alllog
     " SIGTERM SIGINT SIGFPE SIGBUS
```

#### Here's what shellcheck currently says:
```
Line 6	SC1078: Did you forget to close this double quoted string?
Line 7	SC1079: This is actually an end quote, but due to next char it looks suspect.
```
#### Here's what I wanted or expected to see:

Nothing. I believe I am creating a multi-line statement to be executed via the traps defined and that by doubling up the double quotes inside the two outer double quotes, I am escaping the inside ones properly. If I am not, please advise.

**Comment by koalaman** [comment]
Doubling double quotes does not escape them. It just stops one string and starts another, which is a no-op:

```
$ echo "a ""b"" c"
a b c

```

Use `\"` instead:
```
$ echo "a \"b\" c"
a "b" c
```

ShellCheck should probably warn about this though. There's no reason to ever do it.

**Comment by ParadaCarleton** [comment]
> ShellCheck should probably warn about this though. There's no reason to ever do it.

It actually seems to warn when attempting to escape quotes, e.g. `eval "$(starship init \"$SHELL_NAME\")"` creates a warning about not being quoted.
